File: src/Core/Spawning/Spawnee.zig
const sustenet = @import("root").sustenet;
const zig_numerics = sustenet.zig_numerics;

const Vector3 = zig_numerics.vector.Vec3(f32);
const Quaternion = zig_numerics.quaternion.Quaternion(f32);
const Spawnee = @This();

id: u32,
position: Vector3,
rotation: Quaternion,

pub fn new(id: u32, position: Vector3, rotation: Quaternion) Spawnee {
    return Spawnee{
        .id = id,
        .position = position,
        .rotation = rotation,
    };
}

File: rust/src/core/spawning/spawnee.rs
use quaternion::Quaternion;
use vector3::Vector3;

pub struct Spawnee {
    id: u32,
    position: Vector3,
    rotation: Quaternion<f64>,
}

impl Spawnee {
    pub fn new(id: u32, position: Vector3, rotation: Quaternion<f64>) -> Self {
        Spawnee {
            id,
            position,
            rotation,
        }
    }
}
